web/packages/teleterm/src/services/appUpdater/appUpdater.ts

```diff
@@ -89,15 +107,202 @@ export class AppUpdater {
     autoUpdater.forceDevUpdateConfig = true;
 
     this.unregisterEventHandlers = registerEventHandlers(
-      () => {}, //TODO: send the events to the window.
-      () => this.autoUpdatesStatus
+      this.emit,
+      () => this.autoUpdatesStatus,
+      () => this.shouldAutoDownload()
     );
+
+    // Workaround to prevent installing outdated updates.
+    // electron-updater lacks support for this: once an update is downloaded,
+    // it will be installed on quit—even if subsequent update checks report
+    // no new updates.
+    app.on('will-quit', () => {
+      if (!this.isUpdateDownloaded) {
+        // Workaround for Windows and Linux.
+        autoUpdater.autoInstallOnAppQuit = false;
+
+        // macOS-specific workaround:
+        // On macOS, electron-updater downloads the update file and, if
+        // `autoUpdater.autoInstallOnAppQuit` is true, passes it to the native Electron
+        // autoUpdater via a local server. The update is then handed off to the Squirrel
+        // framework for installation (either on demand or after quitting the app).
+        // Unfortunately, once Squirrel gets the update, it is always installed
+        // on quit, regardless of the `autoInstallOnAppQuit` value.
+        // The only workaround I've found is to manually delete the entire Squirrel
+        // update directory for Connect.
+        if (autoUpdater instanceof MacUpdater && app.isPackaged) {
+          const squirrelMacPath = path.join(
+            os.homedir(),
+            'Library',
+            'Caches',
+            'gravitational.teleport.connect.ShipIt'
+          );
+          try {
+            rmSync(squirrelMacPath, {
+              recursive: true,
+              force: true,
+              maxRetries: 2,
+            });
+          } catch {
+            /* empty */
+          }
+        }
+      }
+    });
   }
 
   dispose(): void {
     this.unregisterEventHandlers();
   }
 
+  /**
+   * Checks for app updates.
+   *
+   * This method enhances the standard autoUpdater.checkForUpdates() by adding
+   * the following behaviors:
+   * - It allows update checks during an ongoing download process.
+   * If a new update is found (or no update is available), the current download
+   * is canceled.
+   * - If an update is checked after a download has completed, the updater
+   * automatically transitions to the `update-downloaded` state.
```

Comment:
What is the default behavior for this?

Reply:
If auto download is enabled in Electron Updater, it will attempt to download the update again, so the updater will transition to the `update-downloaded` state too.
But if it's not, then the updater will stay in the `update-available` state.

I clarified that only that second case has been changed.